crates/swc_ecma_transforms_compat/tests/es2017_async_to_generator.rs

```diff
@@ -1707,28 +1707,26 @@ function s(x) {
 }
 function _s() {
   _s = _asyncToGenerator(function* (x, ...args) {
-      var _this1 = this;
       let t = function () {
           var _ref1 = _asyncToGenerator(
-              (function* (y, a) {
-                  var _this = this;
+              function* (y, a) {
                   let r = function () {
                       var _ref = _asyncToGenerator(
-                          (function* (z, b, ...innerArgs) {
+                          function* (z, b, ...innerArgs) {
                               yield z;
                               console.log(this, innerArgs, arguments);
                               return this.x;
-                          }).bind(_this)
-                      ).bind(_this);
```

Comment:
@Austaras @magic-akari Do you have any idea about `.bind(this)`?
`this` in the current context is `this` of anonymous function created by `let r = function () {`

Reply:
I believe if r is an arrow function, bind is still needed(for now)

Reply:
Oh, it's fine then

Reply:
@kwonoj Yes, and I fixed it using `FnEnvState` and `FnEnvHoister`.

Reply:
It's good to see that this issue has been fixed :)

Reply:
guess we can close #2716 even?

Reply:
Not yet, https://github.com/swc-project/swc/blob/d03824b13d8d744f2b16698dbed182e3afe88374/crates/swc_ecma_transforms_compat/src/es2017/async_to_generator.rs#L135 is the last one.